server/audit_math/suite.py

```diff
@@ -1,718 +0,0 @@
-    if n1_original != 0:
-        o1_rate = cvr_stratum.misstatements[(winner, loser)]["o1"] / n1_original
-        o2_rate = cvr_stratum.misstatements[(winner, loser)]["o2"] / n1_original
-        u1_rate = cvr_stratum.misstatements[(winner, loser)]["u1"] / n1_original
-        u2_rate = cvr_stratum.misstatements[(winner, loser)]["u2"] / n1_original
-
-    n1 = math.ceil(n_ratio * n)
-    n2 = int(n - n1)
-
-    if (n1 < n1_original) or (n2 < n2_original):
-        return 1.0
-
-    o1 = (
-        math.ceil(o1_rate * (n - n1_original))
-        + cvr_stratum.misstatements[(winner, loser)]["o1"]
-    )
-    o2 = (
-        math.ceil(o2_rate * (n - n1_original))
-        + cvr_stratum.misstatements[(winner, loser)]["o2"]
-    )
-    u1 = (
-        math.floor(u1_rate * (n - n1_original))
-        + cvr_stratum.misstatements[(winner, loser)]["u1"]
-    )
-    u2 = (
-        math.floor(u2_rate * (n - n1_original))
-        + cvr_stratum.misstatements[(winner, loser)]["u2"]
-    )
-
-    # Because this is a hypothetical sample, we create a
-    # corresponding hypothetical stratum
-    hyp_sample_size = n1
-
-    hyp_misstatements: MISSTATEMENTS = {
-        (winner, loser): {"o1": o1, "o2": o2, "u1": u1, "u2": u2,}
-    }
-
-    hyp_cvr_stratum = BallotComparisonStratum(
-        cvr_stratum.num_ballots,
-        cvr_stratum.vote_totals,
-        hyp_misstatements,
-        hyp_sample_size,
-    )
-
-    # Set up the no-CVR stratum, assuming the sample looks like the
-    # prior round
-    prev_sample_size = bp_stratum.sample_size
-    hyp_sample: BallotPollingStratum.SAMPLE_RESULTS = {"hyp_round": {}}
-
-    cumulative_sample = bravo.compute_cumulative_sample(bp_stratum.sample)
-
-    # Add fake ballots to the hypothetical sample:
-    if prev_sample_size == 0:
-        # If no ballots have been sampled, assume the sample is roughly
-        # the margin
-        hyp_sample["hyp_round"] = {
-            winner: min(
-                int(n2 * (bp_stratum.vote_totals[winner] / bp_stratum.num_ballots)),
-                bp_stratum.vote_totals[winner],
-            ),
-            loser: min(
-                math.ceil(
-                    n2 * (bp_stratum.vote_totals[loser] / bp_stratum.num_ballots)
-                ),
-                bp_stratum.vote_totals[loser],
-            ),
-        }
-    else:
-        # Otherwise use the sample we've seen so far
-        hyp_sample["hyp_round"] = {
-            winner: min(
-                int(
-                    (n2 - prev_sample_size)
-                    * (cumulative_sample[winner] / prev_sample_size)
-                ),
-                bp_stratum.vote_totals[winner],
-            ),
-            loser: min(
-                math.ceil(
-                    (n2 - prev_sample_size)
-                    * (cumulative_sample[loser] / prev_sample_size)
-                ),
-                bp_stratum.vote_totals[loser],
-            ),
-        }
```

Comment:
my reading of the original code is that they create this expected new sample results and add it to the observed sample so far: https://github.com/pbstark/CORLA18/blob/master/code/suite_tools.py#L639-L641

so here i think that would mean taking these values and adding `cumulate_sample[winner or loser]` to them, right?

Reply:
oh wait, nevermind, this is just for the hypothetical new round. ignore